web/handlers/router.go

```diff
@@ -152,41 +153,78 @@ func (r Router) Handler() (http.Handler, error) {
+		Logger:                     r.Logger,
+	})
+
 	router.Handle("/{owner}/{repo}/{branch}/{build:[0-9]+}", &PipelineHandler{
-		PAInterface:                r.PAInterface,
+		PAInterfaceFactory:         r.PAInterfaceFactory,
+		DefaultJXNamespace:         r.DefaultJXNamespace,
 		StoredPipelinesURLTemplate: archivedPipelinesURLTemplate,
 		BuildLogsURLTemplate:       archivedLogsURLTemplate,
 		Render:                     r.render,
 		Logger:                     r.Logger,
 	})
 
+	router.Handle("/ns-{namespace}/{owner}/{repo}/{branch}/{build:[0-9]+}.yaml", &PipelineHandler{
+		PAInterfaceFactory:         r.PAInterfaceFactory,
+		DefaultJXNamespace:         r.DefaultJXNamespace,
+		StoredPipelinesURLTemplate: archivedPipelinesURLTemplate,
+		BuildLogsURLTemplate:       archivedLogsURLTemplate,
+		RenderYAML:                 true,
+		Render:                     r.render,
+		Logger:                     r.Logger,
+	})
+
 	router.Handle("/{owner}/{repo}/{branch}/{build:[0-9]+}.yaml", &PipelineHandler{
-		PAInterface:                r.PAInterface,
+		PAInterfaceFactory:         r.PAInterfaceFactory,
+		DefaultJXNamespace:         r.DefaultJXNamespace,
 		StoredPipelinesURLTemplate: archivedPipelinesURLTemplate,
 		BuildLogsURLTemplate:       archivedLogsURLTemplate,
 		RenderYAML:                 true,
 		Render:                     r.render,
 		Logger:                     r.Logger,
 	})
 
+	router.Handle("/ns-{namespace}/{owner}/{repo}/{branch}/{build:[0-9]+}/logs", &LogsHandler{
+		PAInterfaceFactory:   r.PAInterfaceFactory,
+		DefaultJXNamespace:   r.DefaultJXNamespace,
+		BuildLogsURLTemplate: archivedLogsURLTemplate,
+		Logger:               r.Logger,
+	})
+
 	router.Handle("/{owner}/{repo}/{branch}/{build:[0-9]+}/logs", &LogsHandler{
-		PAInterface:          r.PAInterface,
+		PAInterfaceFactory:   r.PAInterfaceFactory,
+		DefaultJXNamespace:   r.DefaultJXNamespace,
 		BuildLogsURLTemplate: archivedLogsURLTemplate,
 		Logger:               r.Logger,
 	})
 
+	router.Handle("/ns-{namespace}/{owner}/{repo}/{branch}/{build:[0-9]+}/logs/live", &LiveLogsHandler{
+		DefaultJXNamespace: r.DefaultJXNamespace,
+		KubeClient:         kClient,
+		JXClient:           jxClient,
+		TektonClient:       tknClient,
+		Broker:             sse.NewBroker(nil),
+		Logger:             r.Logger,
+	})
+
 	router.Handle("/{owner}/{repo}/{branch}/{build:[0-9]+}/logs/live", &LiveLogsHandler{
-		KubeClient:   kClient,
-		JXClient:     jxClient,
-		TektonClient: tknClient,
-		Namespace:    r.Namespace,
-		Broker:       sse.NewBroker(nil),
-		Logger:       r.Logger,
+		DefaultJXNamespace: r.DefaultJXNamespace,
+		KubeClient:         kClient,
+		JXClient:           jxClient,
+		TektonClient:       tknClient,
+		Broker:             sse.NewBroker(nil),
+		Logger:             r.Logger,
 	})
 
 	router.Handle("/namespaces/{namespace}/pipelineruns/{pipelineRun}", &PipelineRunHandler{
 		TektonClient:                  tknClient,
-		PAInterface:                   r.PAInterface,
```

Comment:
is there a reason to remove this one?

Reply:
Nope. Thanks for pointing out this!

Fixed.

```bash
    ~  curl -vvv http://localhost:8090/namespaces/jx-test1/pipelineruns/oot-example-project-main-release-c54p5                                                                                                                                                                     ✔ 
*   Trying 127.0.0.1:8090...
* Connected to localhost (127.0.0.1) port 8090 (#0)
> GET /namespaces/jx-test1/pipelineruns/oot-example-project-main-release-c54p5 HTTP/1.1
> Host: localhost:8090
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 301 Moved Permanently
< Content-Type: text/html; charset=utf-8
< Location: /ns-jx-test1/keskad/springboot-example-project/main/1
< Date: Mon, 14 Mar 2022 20:53:48 GMT
< Content-Length: 88
< 
<a href="/ns-jx-test1/keskad/springboot-example-project/main/1">Moved Permanently</a>.
```